core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -457,6 +444,31 @@ private void handleProvisionResult(TransferProcess transferProcess, List<Provisi
         }
     }
 
+    private boolean handleProvisionDataAddressResource(ProvisionedDataAddressResource resource, ProvisionResponse response, TransferProcess transferProcess) {
```

Comment:
This method provides side effects, probably I would avoid to have something that returns `boolean` (and looks like a "query" method) and that changes the state at the same time.
Since a `false` result will cause `transitionError` I would put that method call right there (avoiding `store`, because it will be called anyway in the block at line 431), and making the method be a `void` one

Reply:
Looking at this, I think the issue was the method was doing two separate things. I pulled the secret storing up and integrated that with fatal errors, which allowed error transitioning to be handled by a single code path.

Let me know what you think.